pkg/notes/notes_map.go

```diff
@@ -0,0 +1,174 @@
+	notemaps := []ReleaseNotesMap{}
+	yamlReader, err := os.Open(mapPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "opening maps")
+	}
+
+	decoder := yaml.NewDecoder(yamlReader)
+	noteMap := ReleaseNotesMap{}
+
+	for {
+		if err := decoder.Decode(&noteMap); err != nil {
+			//return nil, errors.Wrap(err, "while unmarshaling map yaml")
+			break
+		}
+		logrus.Info("Nota en %s", mapPath)
+		notemaps = append(notemaps, noteMap)
+	}
+
+	// logrus.Infof("Note Value:\n%+v", notemap)
+	return &notemaps, nil
+}
+
+// ReleaseNotesMap Represents
+type ReleaseNotesMap struct {
+	// Pull request where the note was published
+	PR int `json:"pr"`
+	// SHA of the notes commit
+	Commit      string `json:"commit"`
+	ReleaseNote struct {
+		// Text is the actual content of the release note
+		Text *string `json:"text"`
+
+		// Docs is additional documentation for the release note
+		Documentation *[]*Documentation `json:"documentation,omitempty"`
+
+		// Author is the GitHub username of the commit author
+		Author *string `json:"author"`
+
+		// Areas is a list of the labels beginning with area/
+		Areas *[]string `json:"areas,omitempty"`
+
+		// Kinds is a list of the labels beginning with kind/
+		Kinds *[]string `json:"kinds,omitempty"`
+
+		// SIGs is a list of the labels beginning with sig/
+		SIGs *[]string `json:"sigs,omitempty"`
+
+		// Indicates whether or not a note will appear as a new feature
+		Feature *bool `json:"feature,omitempty"`
+
+		// ActionRequired indicates whether or not the release-note-action-required
+		// label was set on the PR
+		ActionRequired *bool `json:"action_required,omitempty"`
+
+		// Tags each note with a release version if specified
+		// If not specified, omitted
+		ReleaseVersion *string `json:"release_version,omitempty"`
+	} `json:"release-note"`
+
+	DataFields map[string]ReleaseNotesDataField `json:"datafields"`
+}
+
+// ReleaseNotesDataField extra data added to a release note
+type ReleaseNotesDataField interface{}
+
+// DirectoryMapProvider is a provider that gets maps from a directory
+type DirectoryMapProvider struct {
+	Path string
+	Maps map[int]map[int]*ReleaseNotesMap
+}
+
+// readMaps Open the dir and read dir notes
+func (mp *DirectoryMapProvider) readMaps() error {
+
+	var fileList []string
+	mp.Maps = map[int]map[int]*ReleaseNotesMap{}
+
+	err := filepath.Walk(mp.Path, func(path string, info os.FileInfo, err error) error {
+		if filepath.Ext(path) == ".yaml" || filepath.Ext(path) == ".yml" {
+			fileList = append(fileList, path)
+		}
+		return nil
+	})
+
+	counter := 0
+	for _, fileName := range fileList {
+		notemaps, err := ParseReleaseNotesMap(fileName)
+		if err != nil {
+			logrus.Warnf("while reading parsing note at %s", fileName)
+			continue
+		}
+		for _, notemap := range *notemaps {
+			if _, ok := mp.Maps[notemap.PR]; !ok {
+				mp.Maps[notemap.PR] = map[int]*ReleaseNotesMap{}
+			}
+			mp.Maps[notemap.PR][len(mp.Maps[notemap.PR])] = &notemap
+			counter++
+		}
+
+		fmt.Printf("%+v", mp)
```

Comment:
This can be removed, right?

Reply:
Yes, sorry. I removed another test output too.